app/bundles/LeadBundle/EventListener/CampaignSubscriber.php

```diff
@@ -351,7 +360,30 @@ public function onCampaignTriggerCondition(CampaignExecutionEvent $event)
             return $event->setResult(false);
         }
 
-        if ($event->checkContext('lead.tags')) {
+        if ($event->checkContext('lead.device')) {
+            $deviceRepo = $this->leadModel->getDeviceRepository();
+            $result     = false;
+
+            $deviceType   = $event->getConfig()['device_type'];
+            $deviceBrands = $event->getConfig()['device_brand'];
+            $deviceOs     = $event->getConfig()['device_os'];
+
+            if (!empty($deviceType)) {
```

Comment:
This execution doesn't really match the UI. I would expect that if I chose a type and an OS, the query would be based on both. But this is bypassing the others once one returns true.